daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/InfosetInputter.scala

```diff
@@ -121,6 +121,13 @@ abstract class InfosetInputter
    * NonTextFoundInSimpleContentException. If the element does not have any
    * simple content, this should return either null or the empty string.
    */
+  def getSimpleText(primType: NodeInfo.Kind, runtimeProperties: Map[String,String]): String =
+    getSimpleText(primType)
+
+  /**
+   * See getSimpleText(primType, runtimeProperties), which has a default
+   * implementation to call this function without the runtimeProperties Map
+   */
   def getSimpleText(primType: NodeInfo.Kind): String
```

Comment:
I question the need to overload `getSimpleText` with one and two parameters.  Wouldn't it be easier to add a default argument to getSimpleText instead?

```scala
def getSimpleText(primType: NodeInfo.Kind, runtimeProperties: Map[String,String] = Map.empty[String, String]): String
```

Passing `Map.empty` as a default argument removes the need to worry about handling `null` or `Option` or auto-converting `Map` to `Option[Map]` (see StackOverflow on [optional parameters](https://stackoverflow.com/questions/4199393/are-options-and-named-default-arguments-like-oil-and-water-in-a-scala-api)).

The corresponding Java API still would need to overload `getSimpleText` with one and two parameters since Java doesn't have default parameters, but you still could simplify the corresponding Scala API the same way.

Reply:
I was trying to prevent any issues with backwards compatibility.

Thinking about it though, this InfosetInputter.scala stuff is "internal" API, so I think it shouldn't be a problem to have this use a default parameter. In fact, since this is internal, this really can drop the single-arg version entirely--Daffodil internals will never call this single-arg version. This does require updating all of our built-in infoset inputters to use the two-arg function, but that's trivial to just add the new parameter.

But I think the public Java/Scala API infoset inputters still need the changes I've added to Infoset.scala. Otherwise, we'll break existing InfosetInputters that don't implement the two arg version. I think...

Note that we also need to take a look at our InfosetOutputter/Inputter API's (see [Daffodil1909](https://issues.apache.org/jira/browse/DAFFODIL-1909)), so we'll probably need to break things at some point. So maybe we break things for that change?

Reply:
I'm fine with changing the internal InfosetInputter API and all the calls to use two arguments (no default parameter needed or used), although that's more work for you.  For backwards compatibility, I agree the public Java/Scala API InfosetInputters still need to use overloaded one-arg and two-arg functions (Java) or a default parameter (Scala), although a default parameter still breaks binary compatibility but not source compatibility (if that matters).  

Did you see my questions about the asciidoc design document and the custom XML InfosetInputter/Outputter for embedded XML strings in binary data too?